astrosat/safers-dashboard-app
# Code linting

Okay, this is a bit of a beast of a PR, it touches every single `*.js` file.

## What have I done

I've fixed the linting and formatting so that it actually works, a lot of it was there already, but it just didn't work. I added in the eslint and prettier config files, along with config for VS Code, so you see the linting errors and fix them on **paste** and **save** actions.

I opened every `*.js` file and saved them, so as to fix formatting. I then went through the outstanding errors and fixed each of those. Things like:

- switching `==` for `===` everywhere.
- Organize imports
- Fix missing hook dependencies, I feel not everyone is clear on these, so I think a little refresher by our devs is a good idea. We can all do with a review of our knowledge from time to time.
- Stop using `indices` as keys for mapped JSX, you should never ever ever do this.....

## How to test

There are too many file changes to expect reviewers to look at the code. Most of it is formatting, or re-arranging imports, therefore too easy to miss real changes. I suggest that instead of looking at code, checkout the branch and run locally. You will see a lack of errors/warnings in the terminal, then use the app, play with each area, search for content, change date ranges, request map requests etc. Basically, check I've not regressed the app in any way.

## Implications of merging this code

If you have a development branch, you will need to `merge` this in and then check for any conflicts. As long as you haven't edited 20 files, this shouldn't be too hard. I'm available to help though.

From now on, you should get linting errors right in your IDE, so you should be able to fix them as you code. I'm still seeing people introduce new errors in the browser console as well, we all need to check we are not adding new ones and that can only be done by testing our own code, before committing it.
